File: epi_judge_cpp_solutions/making_change.cc
#include <array>

#include "test_framework/generic_test.h"

using std::array;

int ChangeMaking(int cents) {
  const array<int, 6> kCoins = {100, 50, 25, 10, 5, 1};
  int num_coins = 0;
  for (int coin : kCoins) {
    num_coins += cents / coin;
    cents %= coin;
  }
  return num_coins;
}

int main(int argc, char* argv[]) {
  std::vector<std::string> args{argv + 1, argv + argc};
  std::vector<std::string> param_names{"cents"};
  return GenericTestMain(args, "making_change.cc", "making_change.tsv",
                         &ChangeMaking, DefaultComparator{}, param_names);
}

File: epi_judge_java_solutions/epi/MakingChange.java
package epi;

import epi.test_framework.EpiTest;
import epi.test_framework.GenericTest;

public class MakingChange {
  @EpiTest(testDataFile = "making_change.tsv")

  public static int changeMaking(int cents) {

    final int[] COINS = {100, 50, 25, 10, 5, 1};
    int numCoins = 0;
    for (int i = 0; i < COINS.length; i++) {
      numCoins += cents / COINS[i];
      cents %= COINS[i];
    }
    return numCoins;
  }

  public static void main(String[] args) {
    System.exit(
        GenericTest
            .runFromAnnotations(args, "MakingChange.java",
                                new Object() {}.getClass().getEnclosingClass())
            .ordinal());
  }
}
